File: suites/02_numerical_computation/comparisons/matrix_multiply.cpp
#include <cstdio>
static long long matrix_checksum(long long size) {
    long long sum = 0;
    for (long long row = 0; row < size; ++row) {
        for (long long col = 0; col < size; ++col) {
            long long cell = 0;
            for (long long inner = 0; inner < size; ++inner) {
                cell += ((row * inner) + 3) * ((inner * col) + 5);
            }
            sum += cell % 104729;
        }
    }
    return sum;
}
int main() { printf("%lld\n", matrix_checksum(64)); return 0; }

File: suites/02_numerical_computation/comparisons/matrix_multiply.rs
fn matrix_checksum(size: i64) -> i64 {
    let mut sum: i64 = 0;
    for row in 0..size {
        for col in 0..size {
            let mut cell: i64 = 0;
            for inner in 0..size {
                cell += ((row * inner) + 3) * ((inner * col) + 5);
            }
            sum += cell % 104729;
        }
    }
    sum
}
fn main() { println!("{}", matrix_checksum(64)); }
